### PR TITLE
Add keywords search

### PR DESCRIPTION
**Pull request by subinasr**
- Addresses #250 

## Changes

* Add keyword search filter in combined indicators page

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

### CODE REVIEW

File: app/views/Dashboard/CombinedIndicators/TopicCard/OutbreakIndicators/index.tsx [hunk 1]
```diff
@@ -31,8 +36,21 @@ function OutbreakIndicators(props: Props) {
...
         handleIndicatorClick,
+        includedIndicators,
+        includedSubvariables,
+        searchText,
     } = props;
 
+    const filteredIndicatorsList = useMemo(() => (
+        list.filter(
+            (indicator) => includedIndicators?.includes(indicator?.indicatorId ?? undefined),
+        ).filter((sub) => includedSubvariables?.includes(sub?.subvariable ?? undefined))
```
Comment (AdityaKhatri) [inline app/views/Dashboard/CombinedIndicators/TopicCard/OutbreakIndicators/index.tsx]:
```suggestion
            (indicator) => includedIndicators?.includes(indicator?.indicatorId ?? undefined) && includedSubvariables?.includes(indicator?.subvariable ?? undefined))
```

File: app/views/Dashboard/CombinedIndicators/TopicCard/index.tsx [hunk 2]
```diff
@@ -20,12 +21,15 @@ const subvariableKeySelector = (d: { emergency: string }) => d.emergency;
...
     topicName: string;
-    topicDescription?: string;
+    topicDescription: string | undefined;
     indicators: IndicatorType[] | undefined;
     showRegionalValue: boolean;
     filterValues: FilterType | undefined;
     setFilterValues: React.Dispatch<React.SetStateAction<FilterType | undefined>>;
     setActiveTab: React.Dispatch<React.SetStateAction<TabTypes | undefined>>;
+    includedIndicators: (string | undefined)[] | undefined;
+    includedSubvariables: (string | undefined)[] | undefined;
```
Comment (AdityaKhatri) [inline app/views/Dashboard/CombinedIndicators/TopicCard/index.tsx]:
```suggestion
    includedIndicators: string[] | undefined;
    includedSubvariables: string[] | undefined;
```

Filter the undefined ones before sending them here.

File: app/views/Dashboard/CombinedIndicators/index.tsx [hunk 3]
```diff
@@ -155,12 +158,16 @@ export type IndicatorType = NonNullable<TopicIndicatorType['indicators']>[number
...
+    thematicDescription: string | undefined;
     indicators: TopicIndicatorType[] | undefined;
     showRegionalValue: boolean;
     filterValues: FilterType | undefined;
     setFilterValues: React.Dispatch<React.SetStateAction<FilterType | undefined>>;
     setActiveTab: React.Dispatch<React.SetStateAction<TabTypes | undefined>>;
+    includedTopics: (string | undefined)[] | undefined;
+    includedIndicators: (string | undefined)[] | undefined;
+    includedSubvariables: (string | undefined)[] | undefined;
+    searchText: string | undefined;
```
Comment (AdityaKhatri) [inline app/views/Dashboard/CombinedIndicators/index.tsx]: same as above

**Review by shreeyash07 — APPROVED** [review]

File: app/views/Dashboard/CombinedIndicators/index.tsx [hunk 4]
```diff
@@ -317,6 +357,68 @@ function CombinedIndicators(props: Props) {
...
+        const indicatorItem = indicatorsWithConcatenatedString?.filter((item) => (
+            (keywordSearchText?.length ?? 0) > 0
+                ? caseInsensitiveSubmatch(item.textToSearch, keywordSearchText)
+                : true
+        ));
+        return ([
+            indicatorItem?.map((item) => item.thematicName),
+            indicatorItem?.map((item) => item.topicName),
+            indicatorItem?.map((item) => item.indicatorId ?? undefined),
+            indicatorItem?.map((item) => item.subvariableId ?? undefined),
```
Comment (AdityaKhatri) [inline app/views/Dashboard/CombinedIndicators/index.tsx]: filter the undefined here

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

**Pull request merged**